public/app/features/logs/logsModel.ts

```diff
@@ -216,8 +216,27 @@ export function dataFrameToLogsModel(
   absoluteRange?: AbsoluteTimeRange,
   queries?: DataQuery[]
 ): LogsModel {
+  // Until nanosecond precision for requests is supported, we need to account for possible duplicate rows.
+  let infiniteScrollingResults = false;
+  queries = queries?.map(query => {
+    if (query.refId.includes('infinite-scroll')) {
+      infiniteScrollingResults = true;
+      return {
+        ...query,
+        refId: query.refId.replace('infinite-scroll-', ''),
```

Comment:
Maybe it makes sense to export the prefix `infinite-scroll-` and just use that one variable all the time?

Reply:
Done.